pkg/services/ngalert/schedule/schedule.go

```diff
@@ -440,18 +464,8 @@ func (sch *schedule) ruleRoutine(grafanaCtx context.Context, key ngmodels.AlertR
 			sch.alertsSender.Send(ctx, key, alerts)
 		}
 		sendDuration.Observe(sch.clock.Now().Sub(start).Seconds())
-	}
 
-	retryIfError := func(f func(attempt int64) error) error {
-		var attempt int64
-		var err error
-		for attempt = 0; attempt < sch.maxAttempts; attempt++ {
-			err = f(attempt)
-			if err == nil {
-				return nil
-			}
-		}
-		return err
```

Comment:
I think it would be much simpler to just add time.Sleep here. This would reduce the diff and perhaps will make backports less painful. 
I wonder what is the reason to delete it?

Reply:
We need control over the loop from the main function - sadly, I needed to remove function in favour of using the loop directly.